homeassistant/components/climate/generic_thermostat.py

```diff
@@ -107,6 +107,7 @@ def __init__(self, hass, name, heater_entity_id, sensor_entity_id,
         sensor_state = hass.states.get(sensor_entity_id)
         if sensor_state:
             self._async_update_temp(sensor_state)
+            self._async_control_heating()
```

Comment:
And since this platform has a dependency on switch and platform, everything that is not discovered should also have been set up already.